File: 3788-maximum-unique-subarray-sum-after-deletion/3788-maximum-unique-subarray-sum-after-deletion.cpp
class Solution {
public:
    int maxSum(vector<int>& nums) {
        int mx=INT_MIN;
        int sum =0;
        set<int> s;
        for(int i=0; i<nums.size(); i++){
            mx = max(nums[i],mx);
            if(nums[i]<=0 || s.find(nums[i])!= s.end()) continue;
            sum+= nums[i];
            s.insert(nums[i]);
        }
        return (mx<=0)?mx: sum;
    }
};

File: 3788-maximum-unique-subarray-sum-after-deletion/3788-maximum-unique-subarray-sum-after-deletion.kt
class Solution {
    fun maxSum(nums: IntArray): Int {
        var mx= Int.MIN_VALUE
        var sum =0
        val s = HashSet<Int>()
        for(num in nums){
            mx = maxOf(mx,num)
            if(num<=0 || s.contains(num)) continue 
            sum+=num
            s.add(num)
        }
        return if(mx<= 0) mx else sum
    }
}